paasta_tools/check_chronos_jobs.py

```diff
@@ -170,32 +185,44 @@ def message_for_status(status, service, instance, cluster):
             "See the docs on paasta rerun here:\n"
             "https://paasta.readthedocs.io/en/latest/workflow.html#re-running-failed-jobs for more details."
         ) % {
-            'service': service,
-            'instance': instance,
-            'cluster': cluster,
+            'service': chronos_job_config.service,
+            'instance': chronos_job_config.instance,
+            'cluster': chronos_job_config.cluster,
             'separator': utils.SPACER,
         }
     elif status == pysensu_yelp.Status.UNKNOWN:
-        return 'Last run of job %s%s%s Unknown' % (service, utils.SPACER, instance)
+        return ('Last run of job %s%s%s Unknown' %
+                (chronos_job_config.service, utils.SPACER, chronos_job_config.instance))
     elif status == pysensu_yelp.Status.OK:
-        return 'Last run of job %s%s%s Succeded' % (service, utils.SPACER, instance)
+        return ('Last run of job %s%s%s Succeded' %
+                (chronos_job_config.service, utils.SPACER, chronos_job_config.instance))
     elif status is None:
         return None
     else:
         raise ValueError('unknown sensu status: %s' % status)
 
 
-def job_is_stuck(last_run_iso_time, interval_in_seconds):
+def job_is_stuck(last_run_iso_time, interval_in_seconds, client, job_name):
     """Considers that the job is stuck when it hasn't run on time
 
     :param last_run_iso_time: ISO date and time of the last job run as a string
     :param interval_in_seconds: the job interval in seconds
+    :param client: configured Chronos client
+    :param job_name: Chronos job name
     :returns: True or False
     """
     if last_run_iso_time is None or interval_in_seconds is None:
         return False
-    last_run_datatime = isodate.parse_datetime(last_run_iso_time)
-    return last_run_datatime + timedelta(seconds=interval_in_seconds) < datetime.now(pytz.utc)
+    dt_next_run = isodate.parse_datetime(last_run_iso_time) + timedelta(seconds=interval_in_seconds)
+    dt_now_utc = datetime.now(pytz.utc)
+    if dt_next_run >= dt_now_utc:
+        return False
+    try:
+        expected_runtime = int(client.job_stat(job_name)['histogram']['99thPercentile'])
+    except KeyError:
+        log.warn("Can't get 99thPercentile for %s." % job_name)
```

Comment:
This can happen a lot, no need to warn. I think debug is fine.

Also log "Assuming a runtime of $interval_in_seconds seconds")